### ISSUE TITLE
offline-usage not possible

### ISSUE THREAD

**Issue by k9ert** [coinables/segwitaddress#2]
There are references to the public internet, e.g. here:
https://github.com/coinables/segwitaddress/blob/9c9494b7863086c51be510660c5f4036130d2863/index.html#L51
This prevents pure offline usage. I'd be quite happy to also use this completely offline.

Other then that, thank you very much for this tool :-)

**Comment by coinables** [comment]
Thank you for pointing that out. I'll make the change tomorrow, or feel
free to do a pull request. Just need to save the minified jquery js locally
and then update the reference link in the html.

On Nov 13, 2017 11:09 PM, "Kim Neunert" <notifications@github.com> wrote:

There are references to the public internet, e.g. here:
https://github.com/coinables/segwitaddress/blob/
9c9494b7863086c51be510660c5f4036130d2863/index.html#L51
This prevents pure offline usage. I'd be quite happy to also use this
completely offline.

Other then that, thank you very much for this tool :-)

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/coinables/segwitaddress/issues/2>, or mute the thread

**Comment by k9ert** [comment]
Thank you very much. I might also create a PR but not yet sure when. If you do, i'd suggest not to take the minified version which would make review much easier for paranoid guys. 
Also maybe you want to make a sha256-checksum for files coming from outside which might shortcut the review for less paranoid guys.

**Comment by k9ert** [comment]
I just started to try and i had issues with bootswatch: https://bootswatch.com/3/
I'd prefer to not modify the downloaded files, however there does not seem to be offline-versions of that theme ("readable"). As far as i believe, that would not happen with plain bootstrap.

Any idea how to proceed with this? Would plain bootstrap fine for you?

**Comment by coinables** [comment]
Thats fine or I can just do it as manual CSS the styling isn't too crazy.
Thanks for your help, its been hard finding time lately to do this.

**Comment by k9ert** [comment]
I wasn't able to replace bootstrap.css without modifying the original bootstrap. That's probably acceptable.
However there is a bigger issues: bitpay's fee-level-recommendation-usage.
https://github.com/coinables/segwitaddress/blob/master/index.html#L461

However, bitpay's feerecommendations are pretty stupid. I just downloaded them and the cheapest one seems to be this:
{"level":"superEconomy","feePerKb":71073,"nbBlocks":24}

Honestly, look at the feerecommendations you get from twitter's bitcoin_fees
https://twitter.com/bitcoin_fees/status/933409762145955840

Fastest Fee: 165 sat/byte Hour Fee: 135 sat/byte Day Fee: 15 sat/byte

That's insane. bitpay's recommendation is IMHO by orders of magnitute higher than it has to be.

I'll try to remove it completely but would be happy to have some feedback for doing it.

**Comment by coinables** [comment]
I ran into the same issue yesterday with bootstrap so I started doing the CSS from scratch. 

I've had the same concerns with bitpay, their estimate is too high.  But not orders of magnitudes, note they are recommending in kb vs bytes. Typically I just take a fraction, in this example I cut their recommendation in half. So with the 70 sats/byte example you showed for superEconomy it would send at 35 sats/byte which I feel is appropriate. If you do it lower than that you'll be waiting hours for confirmation.  I used to default fees on another service at 20 satoshis per byte, but I received to many complaints of transactions taking too long to confirm. That's one of the issues with suggesting fees to the user, which is why I left it only a recommendation to the user but ultimately they can type in their fee.
